firecrown/ccl/two_point.py

```diff
@@ -28,10 +29,18 @@ def parse_config(analysis):
     if 'likelihood' in analysis:
         new_keys['likelihood'] = _parse_likelihood(analysis['likelihood'])
 
-    # read data if there is a sacc file
+    # read data if there is a sacc file or read object
     if 'sacc_file' in analysis:
-        sacc_data = sacc.Sacc.load_fits(
-            os.path.expanduser(os.path.expandvars(analysis['sacc_file'])))
+        warnings.warn("sacc_file is deprecated, use sacc_data instead",
+                      DeprecationWarning)
+        analysis['sacc_data'] = analysis['sacc_file']
+    if 'sacc_data' in analysis:
+        if type('sacc_data') == str:
```

Comment:
the syntax here is wrong. type('str') is always str. You need to grab the key in the analysis dict. Also please use `isinstance(s, str)`. This will catch subclasses and inheritance.